src/processing/resources/index.js

```diff
@@ -37,8 +38,12 @@ export async function process (ctx, customCharset) {
             var urlReplacer = getResourceUrlReplacer(ctx);
             var processed   = processors[i].processResource(decoded, ctx, charset, urlReplacer);
 
-            if (processed === null)
+            if (processed === null) {
+                if (processors[i] === pageProcessor)
```

Comment:
This is potentially infinite recursion. We need better mechanism and more clear mechanism to tell `process` reprocess resource with the modified `charset`